toolkit/tools/imagegen/diskutils/diskutils.go

```diff
@@ -354,6 +361,44 @@ func DetachLoopbackDevice(diskDevPath string) (err error) {
 	return
 }
 
+func WaitForLoopbackToDetach(devicePath string, diskPath string) error {
+	if !filepath.IsAbs(diskPath) {
+		return fmt.Errorf("internal error: loopback disk path must be absolute (%s)", diskPath)
+	}
+
+	delay := 100 * time.Millisecond
+	attempts := 5
+	for failures := 0; failures < attempts; failures++ {
```

Comment:
This could be a possible use of the retry.go functions.

Reply:
I did consider it. But there are two calls where it is pointless to retry. So, I decided to write it manually instead.